### PR TITLE
[example] Add C++ retry example

### PR DESCRIPTION
**Pull request by yijiem**

### CODE REVIEW

**Review by veblush — COMMENTED** [review]
Can we have a cmake build?

File: examples/cpp/retry/server.cc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+using helloworld::Greeter;
+using helloworld::HelloReply;
+using helloworld::HelloRequest;
+
+// Logic and data behind the server's behavior.
+class GreeterServiceImpl final : public Greeter::Service {
+ public:
+  Status SayHello(ServerContext* context, const HelloRequest* request,
+                  HelloReply* reply) override {
+    if (++request_counter_ % request_modulo_ != 0) {
```
Comment (veblush) [inline examples/cpp/retry/server.cc:45]: Having some comments on this logic would be helpful.

**Comment by yijiem** [comment]
> Can we have a cmake build?

done

File: examples/cpp/retry/server.cc (hunk 1 above)
Comment (yijiem) [reply examples/cpp/retry/server.cc:45]: done. thanks!

File: examples/cpp/retry/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# Copyright 202 gRPC authors.
```
Comment (veblush) [inline examples/cpp/retry/CMakeLists.txt:1]: 2024?

**Review by veblush — APPROVED** [review]
LGTM! Thanks!

File: examples/cpp/retry/CMakeLists.txt (hunk 2 above)
Comment (yijiem) [reply examples/cpp/retry/CMakeLists.txt]: good catch! I blame the keyboard lol.

**Pull request closed without merging**